app/Traits/ExecuteRemoteCommand.php

```
@@ -46,6 +46,14 @@ public function execute_remote_command(...$commands)
                 }
             }
 
+            // Check for cancellation before executing commands
+            if (isset($this->application_deployment_queue)) {
+                $this->application_deployment_queue->refresh();
+                if ($this->application_deployment_queue->status === \App\Enums\ApplicationDeploymentStatus::CANCELLED_BY_USER->value) {
+                    throw new \RuntimeException('Deployment cancelled by user', 69420);
```

PHPMD (2.15.0)

Missing class import via use statement (line '53', column '31'). (undefined)

(MissingImport)